NN_module/test.f90

```diff
@@ -1,146 +1,213 @@
-    real rho(nzm)
-        !! air density at pressure levels
-
-    real :: tabs(nx, ny, nzm)
-        !! absolute temperature
-
-    ! Input Variables
-
-    ! Fields from beginning of time step used as NN inputs
-    real tabs_i(nx,ny,nzm)
-        !! Temperature
-    real q_i(nx,ny,nzm)
-        !! Non-precipitating water mixing ratio
-    real :: y_in(nx,ny)
-        !! Distance of column from equator (proxy for insolation and sfc albedo)
-    real adz(nzm)
-        !! ratio of the grid spacing to dz for pressure levels
-    != unit s :: dtn
-    real dtn
-        !! current dynamical timestep (can be smaller than dt)
-    real dz
-        !! current dynamical timestep (can be smaller than dt)
-
-
-    real, dimension(nx,ny,30) :: t_rad_rest_tend, &
-                                 t_delta_adv, q_delta_adv, &
-                                 t_delta_auto, q_delta_auto, &
-                                 t_delta_sed, q_delta_sed
-    real, dimension(nx,ny) :: prec_sed
-
-    ! Test NN routines
-
-    call relu(test_array)
-
-    write (*,*) test_array
-
-    call nn_cf_net_init("./NN_weights_YOG_convection.nc", nin, nout, 30)
-    call nn_cf_net_finalize()
-
-
-    ! Test convection flux routines
-
-
-    t = 0.
-    q = 0.4
-    precsfc = 0.
-    prec_xy = 0.
-    rho = 1.
-    tabs = 1.
-    tabs_i = 287.15
-    q_i = 0.2
-    adz = 1.
-    y_in = 1.
-    dz = 1.
-    dtn = 1.
-
-    nrf = 30
-
-    call nn_convection_flux_init("./NN_weights_YOG_convection.nc")
-
-    call nn_convection_flux(tabs_i, q_i, y_in, &
-                            tabs, &
-                            t, q, &
-                            rho, adz, dz, dtn, &
-                            t_rad_rest_tend, &
-                            t_delta_adv, q_delta_adv, &
-                            t_delta_auto, q_delta_auto, &
-                            t_delta_sed, q_delta_sed, prec_sed)
-
-    q(:,:,1:nrf) = q(:,:,1:nrf) + q_delta_adv(:,:,:) &
-                                + q_delta_auto(:,:,:) &
-                                + q_delta_sed(:,:,:)
-    t(:,:,1:nrf) = t(:,:,1:nrf) + t_delta_adv(:,:,:) &
-                                + t_delta_auto(:,:,:) &
-                                + t_delta_sed(:,:,:) &
-                                + t_rad_rest_tend(:,:,:)*dtn
-
-    call nn_convection_flux_finalize()
+module tests
+  !! Module containing individual tests for the CAM ML code
+
+  !--------------------------------------------------------------------------
+  ! Libraries to use
+  use netcdf
+  use nn_cf_net_mod, only: relu, net_forward, nn_cf_net_init, nn_cf_net_finalize
+  use nn_convection_flux_mod, only:   nn_convection_flux, nn_convection_flux_init, nn_convection_flux_finalize
+  use test_utils, only: assert_array_equal
+
+  implicit none
+
+  character(len=15) :: pass = char(27)//'[32m'//'PASSED'//char(27)//'[0m'
+  character(len=15) :: fail = char(27)//'[31m'//'FAILED'//char(27)//'[0m'
+
+  real(4), dimension(148) :: nn_out_ones
```

Comment:
make 148 a parameter?

Reply:
This should be resolved in the latest commit.